### PR TITLE
Use EnumerableUtils to get filter

### PR DESCRIPTION
**Pull request by MattMSumner**

### CODE REVIEW

**Comment by stefanpenner** [comment]
`map` may also be one you wish to use this way, as it is es5+ only

File: addon/adapters/firebase.js [hunk 1]
```diff
@@ -323,27 +325,26 @@ export default DS.Adapter.extend(Ember.Evented, {
...
     });
 
     // Dirty
-    dirtyRecords = ids.filter(function(id) {
+    dirtyRecords = filter(ids, function(id) {
       var type = relationship.type;
       return store.hasRecordForId(type, id) && store.getById(type, id).get('isDirty') === true;
     });
-    dirtyRecords = Ember.A(dirtyRecords.concat(addedRecords)).uniq().map(function(id) {
+    dirtyRecords = map(Ember.A(dirtyRecords.concat(addedRecords)).uniq(), function(id) {
```
Comment (MattMSumner) [inline addon/adapters/firebase.js]: Is there another way to handle `.uniq` here? cc @stefanpenner

**Comment by MattMSumner** [comment]
Is the goal here to remove the use of `Ember.A` entirely? This is to handle applications where extend prototypes is false.

**Comment by MattMSumner** [comment]
Closing in favor of https://github.com/firebase/emberfire/pull/190

**Pull request closed without merging**